common/config.js

```diff
@@ -0,0 +1,36 @@
+(function() {
+    'use strict';
+
+    angular.module('chaise.config', ['ermrestjs'])
+
+    .run(['appName', 'ConfigUtils', 'ERMrest', 'headInjector', 'MathUtils', 'UriUtils', '$rootScope', '$window', function(appName, ConfigUtils, ERMrest, headInjector, MathUtils, UriUtils, $rootScope, $window) {
+        headInjector.setWindowName();
+        // initialize dcctx object
+        $window.dcctx = {
+            cid: appName,
+            pid: MathUtils.uuid(),
+            wid: $window.name
+        }
+        // set chaise configuration based on what is in `chaise-config.js` first
+        ConfigUtils.setConfigJSON();
+
+        ERMrest.onload().then(function () {
+            var cc = ConfigUtils.getConfigJSON(),
+                service = cc.ermrestLocation,
+                catalogId = UriUtils.getCatalogId();
+
+            if (catalogId) {
+                ERMrest.ermrestFactory.getServer(service, { cid: dcctx.cid, pid: dcctx.pid, wid: dcctx.wid }).catalogs.get(catalogId).then(function (response) {
```

Comment:
dcctx is not defined

Reply:
It technically is. Anything that is defined on `$window` (or in js, `window`), is able to be accessed as if it were defined in the current scope. I can change it so it's clear that we are fetching it from the `$window` object by adding a variable declaration in here.

Reply:
oh right. I didn't pay attention to that. I think it's better if you explicitly use `$window.dcctx` to be aligned with the file lines above it that you're attaching the value to `$window` object.